### ISSUE TITLE
Logging book drop? 

### ISSUE THREAD

**Issue by Tempest-Breakerofthings** [Arx-Game/arxcode#122]
**Is your feature request related to a problem? Please describe.**
A common problem seeming to appear in the archives is the book drop object we have for purposes of people dropping off texts to be archived seems to be mysteriously raided.

**Describe the solution you'd like**
An object which records details of the date, time and names of individuals who remove or add objects and the name of the object itself. 

**Describe alternatives you've considered**
Doing away with book drops entirely and manually having someone responsible for messengered copies of books for libraries. 

**Additional context**

**Comment by TehomCD** [comment]
Non-repudiation via container logs seems like a reasonable idea. Though I'm tempted to have a new type of library container where it stores the templates of books and destroys/creates book objects by demand.

**Comment by Tempest-Breakerofthings** [comment]
ooooooH That's brilliant! Though you'd need to find a way to make the template title show in that case, that's the only issue with templates, the title coloring has to be maintained separately.  Alternatively granting the ability to give entire orgs template accesses or transfer ownership of a template

**Comment by rozzingit** [comment]
Idk if it's just me but I'd find the idea of giving people template access to a book I wrote just to be able to donate it to a library to be...iffy? A bit. Like, if I'm giving the Archive a book for them to keep, I'm not giving them the right to copy it however they like.

The thought that came to mind to me was having a dropbox container type that you could put things into but not remove them while it's locked. I was about to say "but that seems excessive for what seems to be a pretty limited problem," but, I mean, if you're already talking about a container to store templates, maybe it's not excessive!

**Comment by Tempest-Breakerofthings** [comment]
The idea behind being able to assign template access is something separate from donating to the library.  For example, if I write a book IC and I want to be able to give a whole org to create copies for it, I'd like to have that ability as well, rather than just manually assigning names to the template.  Its something separate from the logging bookcase, but came to mind in part because of it.  If you don't want the archive to copy a book however they like do a private template.  Just bear in mind if that character dies no one is able to make copies. 

The current issue right now is there are literally dozens of books that have multiple copies.    Ideally, all books that have multiple copies should be done as +template though most people don't know how yet.   The idea is that it works something like a hyperlink, preventing the storage of duplicate data. 

Being able to transfer ownership completely would eliminate the problem of someone else creating the template for you.  (at least if I've understood the concept correctly.) 

I love the locked cases, but I also love the idea of a case that just stores the template data and creates and destroys as necessary.  

Hopefully that made sense.

**Comment by rozzingit** [comment]
Ah, yeah, adding a functionality to a template to give access to a whole org would be legit! I do get the idea that if a PC dies with their templates private that no one can make copies. I know that I personally prefer that? Like I had a writer PC and he's not dead but he's inactive and I don't actually want other people making copies of his stuff with him gone. BUT I imagine that staff would let you flip your templates public if your PC died if you wanted to. Mostly I was arguing against the idea of requiring giving a book to a library meaning giving up template access for it. When I've donated books, I just gave up a copy and expected them to shelve that copy. So I kind of feel like a dropbox container type would work there -- one that you can drop things into but, while it's locked, can't remove from.

The issue about multiple book copies is one that I think @emonical was going to work on, as the creator of the template code. Mainly I think there was a point when he was working on a script that would convert existing book text into private templates owned by the original book's writer. I imagine that just got put by the wayside just cause, like, time and such. I love templates, and there might be something that could be done with the `write` code to remind people to use them?
